features/cellular/framework/AT/AT_CellularDevice.cpp

```diff
@@ -103,12 +114,15 @@ CellularNetwork *AT_CellularDevice::open_network(FileHandle *fh)
     if (!_network) {
         ATHandler *atHandler = get_at_handler(fh);
         if (atHandler) {
-            _network = new AT_CellularNetwork(*atHandler);
+            _network = open_network_impl(*atHandler);
```

Comment:
General inconsistency here, both in old and new code - you're checking NULL returns from `new`. 

Plain `new` can't return NULL - it generates an exception or on Mbed OS causes a fatal system error.

If you want these error paths, use `new (std::nothrow)`

Reply:
Ah good point, I will create a ticket about checking other places were new is used in our cellular code. Removed unnecessary if's.